### DISCUSSION TITLE
Which Z-wave JS UI version changes the MQTT topic structure?

### DISCUSSION THREAD

**Discussion by sigzug** [zwave-js/zwave-js-ui#3660] [Q&A] [answered]
Hello!

I noticed first in this version:

```
zwave-js-ui: 9.9.0.09b107a
zwave-js: 12.4.4
```

that the topic structure in the MQTT broker went from using the CommandClass IDs to a structure using strings of the CommandClasses. 

Ex.:
`zwave/room1/entity_1/37/1/currentValue`

became

`zwave/room1/entity_1/switch_binary/endpoint_1/currentValue`

I couldn't find this change in the patch notes, so I'm wondering if anyone knows when this was changed or if its something else I have done that could have changed this somehow.

The version I use that has the "old" structure is:

```
zwave-js-ui: 9.3.1.f26b79e
zwave-js: 12.2.3
```

Thank you!

**Comment by robertsLando** [comment, marked as answer]
@sigzug This isn't changed it depends on if you are using gateway mode with named topics or valueId topics, check your settings., here are the docs: https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=gateway

**Reply by sigzug** [reply]
Aha, I see. Thanks!
